task3_2/js/function.js

```diff
@@ -0,0 +1,227 @@
+//isArray function
+function isArrayPrototype(param) {
+  return Object.prototype.toString.call(param) === '[object Array]';
+}
+
+function isArrayInsanceof(param) {
+  return param instanceof Array;
+}
+
+function isArrayConstructor(param) {
+  return Boolean(param) && param.constructor === Array; //
+}
+
+function testIsArray(func) {
+  var a = [1,2,3]
+  var b = 3;
+  var c = { name: "name" };
+  var d = [];
+  var str = "sdfsfd";
+  console.log("test iaArray:");
+  console.log(func(a));
+  console.log(func(d));
+  console.log(func(b));
+  console.log(func(c));
+  console.log(func(str));
+}
+
+//range function
+function range(a, b, c) {
+  var arr = [];
+  if (arguments.length == 3) {
+    var trueParam = false;
+    if ((a < b && 0 < c) || (a > b && 0 > c)) {
+      trueParam = true;
+    }
+    while (Math.abs(a) < Math.abs(b) && trueParam) {
+      arr.push(a);
+      a+=c;
+    }
+  } else if (arguments.length == 2) {
+    for (var i = a; i < b; i++) {
+      arr.push(i);
+    }
+  } else if (arguments.length == 1) {
+    for (var i = 0; i < a; i++) {
+      arr.push(i);
+    }
+  }
+  return arr;
+}
+
+function testRange() {
+  console.log("test range:");
+  console.log(range(9, 10, 2));
+  console.log(range(1, 10, -2));
+  console.log(range(1, -10, -2));
+  console.log(range(1, 10, 2));
+  console.log(range(10, 10, 2));
+  console.log(range(11, 10, 2));
+  console.log(range(-1, 10, 22));
+
+  console.log(range(1, 5));
+  console.log(range(5, 1));
+  console.log(range(1, 1));
+
+  console.log(range(5));
+  console.log(range(0));
+  console.log(range(-5));
+}
+
+//compact function
+function compactFor(arr) {
+  if (arr instanceof Array) {
+    var reaArr = [];
+    for (var i = 0; i < arr.length; i++) {
+      if (Boolean(arr[i])) {
+        reaArr.push(arr[i]);
+      }
+    }
+    return reaArr;
+  }
+}
+
+function compactFilter(arr) {
+  if (arr instanceof Array) {
+    return arr.filter(
+      function(number) {
+        return Boolean(number);
```

Comment:
is there big difference in this case `return Boolean(number)` and `return number`?
PS it's not number